mysql/resource_user.go

```diff
@@ -100,10 +101,23 @@ func resourceUser() *schema.Resource {
 				Optional: true,
 				Default:  "NONE",
 			},
+
+			"retain_old_password": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
 		},
 	}
 }
 
+func checkRetainCurrentPasswordSupport(ctx context.Context, meta interface{}) (bool, error) {
```

Comment:
Consider returning just `bool` or just `error`.
It seems you never use the `bool` output, so maybe it is useless?